js/src/App.vue

```diff
@@ -44,6 +62,16 @@ export default {
           // If p is set, this is a ping message that only serves to keep the connection open.
           // Break immediately.
           if (res.p !== undefined) {
+            if (timeoutId) {
+              clearTimeout(timeoutId)
+            }
+
+            timeoutId = setTimeout(() => {
+              // we're disconnected, try to connect again
+              console.warn("Disconnected, try to reconnect a few times")
+              this.reconnect()
+            }, 10000)
+            console.debug("Ping")
```

Comment:
Ping is probably not a great message, but it's accurate, as this will be printed every time a ping is received (`if (res.p !== undefined)`). Perhaps "Received ping"?